Content.Server/CrewManifest/CrewManifestSystem.cs

```diff
@@ -0,0 +1,272 @@
+    /// </summary>
+    /// <param name="station">Station that we're displaying the crew manifest for.</param>
+    /// <param name="session">The player's session.</param>
+    /// <param name="owner">The owner of this EUI, if there was one.</param>
+    public void CloseEui(EntityUid station, IPlayerSession session, EntityUid? owner = null)
+    {
+        if (!HasComp<StationRecordsComponent>(station))
+        {
+            return;
+        }
+
+        if (!_openEuis.TryGetValue(station, out var euis)
+            || !euis.TryGetValue(session, out var eui))
+        {
+            return;
+        }
+
+        if (eui.Owner == owner)
+        {
+            eui.Close();
+            euis.Remove(session);
+        }
+
+        if (euis.Count == 0)
+        {
+            _openEuis.Remove(station);
+        }
+    }
+
+    /// <summary>
+    ///     Builds the crew manifest for a station. Stores it in the cache afterwards.
+    /// </summary>
+    /// <param name="station"></param>
+    private void BuildCrewManifest(EntityUid station)
+    {
+        var iter = _recordsSystem.GetRecordsOfType<GeneralStationRecord>(station);
+
+        if (iter == null)
+        {
+            return;
+        }
+
+        var entries = new CrewManifestEntries();
+
+        foreach (var recordObject in iter)
+        {
+            if (recordObject == null)
+            {
+                continue;
+            }
+
+            var record = recordObject.Value.Item2;
+            var entry = new CrewManifestEntry(record.Name, record.JobTitle, record.JobIcon, record.JobPrototype);
+
+            entries.Entries.Add(entry);
+        }
+
+        if (_cachedEntries.ContainsKey(station))
+        {
+            _cachedEntries[station] = entries;
+        }
+        else
+        {
+            _cachedEntries.Add(station, entries);
+        }
+    }
+}
+
+[AdminCommand(AdminFlags.Admin)]
+public sealed class CrewManifestCommand : IConsoleCommand
+{
+    public string Command => "crewmanifest";
+    public string Description => "Opens the crew manifest for the given station.";
+    public string Help => $"Usage: {Command} <entity uid>";
+
+    public void Execute(IConsoleShell shell, string argStr, string[] args)
+    {
+        if (args.Length != 1)
+        {
+            shell.WriteLine($"Invalid argument count.\n{Help}");
+            return;
+        }
+
+        var entMan = IoCManager.Resolve<IEntityManager>();
+
+        if (!EntityUid.TryParse(args[0], out var uid))
+        {
+            shell.WriteLine($"{args[0]} is not a valid entity UID.");
+            return;
+        }
+
+        if (shell.Player == null || shell.Player is not IPlayerSession session)
+        {
+            shell.WriteLine("You must run this from a client.");
+            return;
+        }
+
+        var crewManifestSystem = entMan.EntitySysManager.GetEntitySystem<CrewManifestSystem>();
+
+        crewManifestSystem.OpenEui(uid, session);
+    }
```

Comment:
Should probably be localised I guess.

Reply:
i feel like a lot of these are generic enough that commands could just have localization strings that aren't specific to one command - might be for another PR unless these already exist?